tests/test_linearize.py

```diff
@@ -0,0 +1,138 @@
+import dataclasses
+from typing import Optional
+
+import pytest
+from hypothesis import given, strategies as st
+from hypothesis.internal.conjecture.data import Status
+
+from hypofuzz.database import (
+    Phase,
+    Report,
+    StatusCounts,
+    WorkerIdentity,
+    linearize_reports,
+)
+
+
+def workers(*, uuid):
+    return st.builds(
+        WorkerIdentity,
+        uuid=st.just(uuid),
+        operating_system=st.sampled_from(["darwin", "emscripten", "linux", "win"]),
+        python_version=st.just("3.12.10"),
+        hypothesis_version=st.just("6.101.2"),
+        hypofuzz_version=st.just("24.04.02"),
+        pid=st.integers(min_value=0),
+        hostname=st.text(st.characters(codec="ascii")),
+        pod_name=st.just(None),
+        pod_namespace=st.just(None),
+        node_name=st.just(None),
+        pod_ip=st.just(None),
+        container_id=st.just(None),
+        git_hash=st.just(None),
+    )
+
+
+@st.composite
+def reports(
+    draw, *, count_workers: Optional[int] = None, overlap: bool = False
+) -> list[Report]:
+    # all of this min_size=len(uuids) etc is going to lead to terrible shrinking.
+    # But the alternative of while draw(st.booleans()) will generate too-small
+    # collections. Use `more` from hypothesis internals?
```

Comment:
Let's decompose this: generate a `worker_and_run()` or something, and then have `reports()` draw a list of those.  We can pass in a list of (db_key, nodeid) pairs for each report to sample from; and have the multi-worker reports strategy add whatever offset we need after the fact if overlap=False (noting that we probably want a smaller upper bound on timestamps).

Reply:
ok if I come back to this one? I definitely want a strong strategy here, but also want to write the overlapping case first before making it smarter